app/core/models_core.py

```diff
@@ -118,6 +125,21 @@ class CoreGroup(Base):
     )
 
 
+class CoreSchool(Base):
+    __tablename__ = "core_school"
+
+    id: Mapped[str] = mapped_column(String, primary_key=True, index=True)
+    name: Mapped[str] = mapped_column(String, index=True, nullable=False, unique=True)
+    email_regex: Mapped[str] = mapped_column(String, nullable=False)
+
+    students: Mapped[list["CoreUser"]] = relationship(
+        "CoreUser",
+        back_populates="school",
+        lazy="selectin",
+        default_factory=list,
+    )
```

Comment:
I don't think this is really useful to index `name`. I think this would only be used to check another school does not exist with the same name, which should not happen often.

I would say we don't need to init the `students` list, but as you prefer

```suggestion
    id: Mapped[PrimaryKey]
    name: Mapped[str] = mapped_column(unique=True)
    email_regex: Mapped[str]

    students: Mapped[list["CoreUser"]] = relationship(
        "CoreUser",
        back_populates="school",
        lazy="selectin",
        init=False,
    )
```